src/ipc/handler.c

```diff
@@ -800,6 +802,153 @@ static int ipc_glb_gdb_debug(uint32_t header)
 
 }
 
+#if CONFIG_PROBE
+static inline int ipc_probe_init(uint32_t header)
+{
+	struct sof_ipc_probe_dma_set_params *params = _ipc->comp_data;
+	int dma_set_up = (params->hdr.size - sizeof(*params)) /
+			 sizeof(struct probe_dma);
```

Comment:
this is a bit odd. If the IPC payload is incorrect, it will be translated not as an error but a feature: there is no extraction DMA.

A more strict handling would be nice, e.g. if there is no extraction DMA it should be explicit

Reply:
@plbossart True. We'll wrap that in a macro of sorts.